### DISCUSSION TITLE
테스트 진행 시 결과값이 ?표로 표시되네요.

### DISCUSSION THREAD

**Discussion by LPro4** [naver/ngrinder#921] [Q&A]
websocket을 테스트하려고 라이브러리를 추가했습니다.
추가하고 나서는 해당 라이브러리를 사용하지 않아도 아래와 같이 결과값 일부가 ?로 표시가 되네요
테스트 자체는 정상적으로 완료가 되고,  성능테스트로 진행해봐도 script error로만 표시되고 별다른 오류로그는 남는게 없네요

![image](https://user-images.githubusercontent.com/57122371/196370759-a6da1b77-51f0-4112-84e3-302195edfe22.png)
![image](https://user-images.githubusercontent.com/57122371/196370554-a1af7c50-4788-4f71-ac42-03c71ad2956c.png)

해결방법을 찾아보려했으나 도저히 방법을 못찾겠어서 문의를 남깁니다.
nGrinder 최신버전으로 업데이트도 해봤으며, 관련된 의존성과 같은 버전으로 사용도 해봤습니다..
참조한 의존성도 같이 첨부드려요.

![image](https://user-images.githubusercontent.com/57122371/196371233-b712a345-534c-4b67-8ab0-78469e1b9b37.png)

감사합니다.

**Comment by junoyoon** [comment]
websocket 을 구현하는 대부분의 라이브러리는  async 기반이라 ngrinder 와 호환되지 않습니다. ngrinder 는 blocking 일때만 정확히 수집을 하므로, 적절한 barrier 를 사용하셔서 websocket 호출을 synchronized 코드로 변경하셔서 테스트하면 됩니다. ngrinder 는 thread local 을 사용하여 각종 정보를 저장합니다.
그러나 구현이 그다지 쉽진 않습니다. 저는 websocket 을 테스트 하시고자 한다면, ngrinder 가 아닌 다른 도구를 사용하시기를 권장 드립니다,.
